src/net/parsers/texttracks/webvtt.js

```diff
@@ -0,0 +1,311 @@
+    styleContent += linified[index];
+    index++;
+  }
+
+  className.forEach(c => {
+    styleElements.push({
+      className: c,
+      styleContent: styleContent.replace(/\s/g,""),
+    });
+  });
+}
+
+/**
+ * 
+ * @param {number} index 
+ * @param {Array<string>} linified 
+ * @param {Array<Object>} styleElements 
+ */
+function parseCue(index, linified, styleElements) {
+  const region = document.createElement("div");
+  const regionAttr = document.createAttribute("style");
+  regionAttr.value = 
+  "width:100%; \
+  height:100%; \
+  display:flex; \
+  flex-direction:column; \
+  justify-content:flex-end; \
+  align-items:center;";
+  region.setAttributeNode(regionAttr);
+
+  // Get Header. It may be a class name associated with cue.
+  const header = linified[index];
+  index++;
+
+  // Get time ranges.
+  const timeCodes = linified[index];
+  const range = parseTimeCode(timeCodes);
+  index++;
+
+  // Get content, format and apply style.
+  const pElement = document.createElement("p");
+  const pAttr = document.createAttribute("style");
+  pAttr.value = "text-align:center";
+  pElement.setAttributeNode(pAttr);
+
+  const spanElement = document.createElement("span");
+  const attr = document.createAttribute("style");
+  attr.value = 
+  "background-color:rgba(0,0,0,0.8); \
+  color:white;";
+  spanElement.setAttributeNode(attr);
+
+  const styles = styleElements
+  .filter(f => f.className === header || f.className === "cue")
+  .map(f => f.styleContent);
+  if(styles) {
+    attr.value += styles.join;
+    spanElement.setAttributeNode(attr);
+  }
+
+  while (!(linified[index].match(/^\r/) || linified[index].length === 0)) {
+
+    if(spanElement.childNodes.length != 0) {
+      spanElement.appendChild(document.createElement("br"));}
+    
+    formatWebVTTtoHTML(linified[index], styleElements)
+      .forEach(child =>{
+        spanElement.appendChild(child);
+      });
+
+    index++;
+  }
+
+  region.appendChild(pElement) ;
+  pElement.appendChild(spanElement);
+
+  return { start: range.start, end: range.end, content: region };
+}
+
+function parseTimeCode(text) {
+  const both = text.split(" --> ");
+
+  const s = both[0]
+        .split(":").map(val => parseFloat(val));
+  const e = both[1].split(" ")[0].split(":")
+        .map(val => parseFloat(val));
+
+  const start = s[2] + s[1] * 60 + s[0] * 60 * 60;
+  const end = e[2] + e[1] * 60 + e[0] * 60 * 60;
+
+  return { start, end };
+}
+ 
+  /**
+  * Format WebVTT tags and classes into usual HTML.
+  * <b *> => <b>
+  * <u *> => <u>
+  * <i *> => <i>
+  * <c.class *> => <c.class>
+  * Style is inserted if associated to tag or class.
+  * @param {string} text 
+  * @param {Array<Object>} styleElements
+  */
+function formatWebVTTtoHTML(text, styleElements){
+
+  const classes = ["U","I","B"];
+  const styleClasses = styleElements.map(f => f.className.toUpperCase());
+  const filtered = text
+    // Remove timestamp tags
+    .replace(/<[0-9]{2}:[0-9]{2}.[0-9]{3}>/, "")
```

Comment:
You can use timestamp tag as this one : 

```
1
00:16.500 --> 00:18.500
When the moon <00:17.500>hits your eye
```

We don't need this style of timestamp declaration, because we can't deal with text appearing progressively in the same cue. 
So let's keep it as a regex filter ? Regex is enough specific not to catch anything that is not a timestamp tag.